addons/account/models/account_tax.py

```diff
@@ -433,6 +433,21 @@ def _parse_name_search(name):
                 list_name[i] = '%'.join(re.sub(r"\W+", "", name))
         return ''.join(list_name)
 
+    @api.model
+    def _search(self, domain, offset=0, limit=None, order=None):
+        """
+        Intercept the search on `name` to allow searching more freely on taxes
+        when using `like` or `ilike`.
+        """
+        def preprocess_name_search(domain_item):
+            if not isinstance(domain_item, (list, tuple)) or domain_item[0] != 'name' or domain_item[1] not in ('ilike', 'like'):
+                return domain_item
+            field, operator, value = domain_item
+            if field == 'name' and operator in ('ilike', 'like'):
+                return field, operator, AccountTax._parse_name_search(value)
```

Comment:
The "domain item" is generally called a leaf, and I think that a pattern matching is more readable in this case 
```suggestion
        def preprocess_name_search(leaf):
            match leaf:
                case ('name', 'ilike'|'like' as operator, str() as value):
                    return ('name', operator, AccountTax._parse_name_search(value)
                case _:
                    return leaf
```

Reply:
Nice, I've never used the `match` construct to be honest, so I'm learning new stuff 💯 
I agree it looks more clear and concise, so changed it.

Reply:
On second thought, this approach limits the leaf to be a `tuple` and not a `list`. So I guess this search will not apply when using e.g. a JSON-RPC call (where you can't send tuples).

Reply:
I could also do something like this instead, but that already becomes pretty bloated maybe?
```py
def preprocess_name_search(leaf):
    match leaf:
        case (
            ('name', 'ilike' | 'like' as operator, str() as value) |
            ['name', 'ilike' | 'like' as operator, str() as value]
        ):
            return ('name', operator, AccountTax._parse_name_search(value))
        case _:
            return leaf
```

Reply:
No it doesn't, it just checks that it can be unpacked to 3 elements